upack/src/upack/Msg.scala

```diff
@@ -128,7 +128,16 @@ object Obj{
 
   def apply(): Obj = Obj(new mutable.LinkedHashMap[Msg, Msg]())
 }
-case class Ext(tag: Byte, data: Array[Byte]) extends Msg
+case class Ext(tag: Byte, data: Array[Byte]) extends Msg {
+  override def equals(other: Any): Boolean = other match {
+    case Ext(tagOther, dataOther) =>
+      tag == tagOther && java.util.Arrays.equals(data, dataOther)
+    case _ => false
+  }
+
+  override def hashCode: Int =
+    MurmurHash3.bytesHash(b, "Ext".hashCode)
```

Comment:
`tag` should be represented in the hash somehow. We could probably just use it as the seed.

Reply:
The more I think about this, the less comfortable I am about updating `equals` and `hashCode` here. That could lead to some end-user changes and it isn't consistent with some of the other types (eg. `Bytes`). Maybe this is still desirable change, but probably not in a PR about fixing ext parsing.

I'm going to switch the round-tripping test I added to assert equivalence of the serialized and serialized-deserialized-reserialized payloads (that way we don't need to call `equals` on a `Msg`)

Reply:
👍 for smaller, well-scoped PRs.

Re: `equals`/`hashCode`, it's hard to imagine that any users are depending on the behavior being broken other than to work around it.

Good point about `case class Binary(value: Array[Byte]) extends Msg` being affected, too. A fix for one should probably fix both.

Reply:
The rabbit hole goes deeper. Given that `0.0d == 0L`, should `Float64(0.0) == Int64(0L)`? Probably so! This is especially relevant since round-tripping `Int64(0L)` through messagepack will come back as `Int32(0)`. I think you're wise to push `equals`/`hashCode` to a separate PR. Making assertions against the `Array[Byte]` sounds good for now.

Reply:
> Given that `0.0d == 0L`, should `Float64(0.0) == Int64(0L)`? 

Personally, I don't think this should be the case - the "int format family" and "float format family" are separate. I further wish that `0.0d == 0L` would fail to typecheck instead of returning true (that's another rabbit whole though, and Scala is unlikely yo ever change that).

> This is especially relevant since round-tripping `Int64(0L)` through messagepack will come back as `Int32(0)`.

I agree `Int64(0L)` should roundtrip, but on the other hand: is there really a need for the distinction? Why not just a single `case class Int(value: Long) extends Msg` (and `UInt` for unsigned)? We don't distinguish `Int16` or the fixint cases, so why distinguish the 32 and 64 bit cases?